client/client.go

```diff
@@ -48,6 +52,10 @@ type Config struct {
 	// BaseURL contains the base URL where snappy daemon is expected to be.
 	// It can be empty for a default behavior of talking over a unix socket.
 	BaseURL string
+
+	// Public determines whether or not this client will talk to the private or
+	// public snapd socket (private by default).
+	Public bool
```

Comment:
I've now gotten rid of this boolean completely. The client now automatically uses the most powerful socket available (as discussed on telegram). Note that this will likely cause an apparmor denial message in hooks that don't use the snapd-control interface.